server/storage/mvcc/index.go

```diff
@@ -38,16 +40,80 @@ type index interface {
 	KeyIndex(ki *keyIndex) *keyIndex
 }
 
+type rangeCache struct {
+	sync.RWMutex
+	data map[string]rangeCacheElem
+}
+
+type rangeCacheElem struct {
+	end   []byte
+	atRev int64
+	total int
+	ts    int64
+}
+
 type treeIndex struct {
 	sync.RWMutex
-	tree *btree.BTree
-	lg   *zap.Logger
+	tree  *btree.BTree
+	cache rangeCache
+	stopc chan struct{}
+	lg    *zap.Logger
 }
 
-func newTreeIndex(lg *zap.Logger) index {
-	return &treeIndex{
+func newTreeIndex(lg *zap.Logger, stopc chan struct{}) index {
+	ti := &treeIndex{
 		tree: btree.New(32),
-		lg:   lg,
+		cache: rangeCache{
+			RWMutex: sync.RWMutex{},
```

Comment:
Not used?

Reply:
Hi Gyuho, thanks for reviewing this change. I am now still working on this and I saw some error of "missing content-type field" in some of the tests. I am trying to locate the root cause now by commenting out some of the code.